packages/dataviews/src/components/dataviews-view-config/index.tsx

```diff
@@ -237,236 +240,331 @@ function ItemsPerPageControl() {
 	);
 }
 
-interface FieldItemProps {
-	id: any;
-	label: string;
-	index: number;
-	isVisible: boolean;
-	isHidable: boolean;
-}
-
 function FieldItem( {
-	field: { id, label, index, isVisible, isHidable },
-	fields,
-	view,
-	onChangeView,
+	field,
+	isVisible,
+	isFirst,
+	isLast,
+	canMove = true,
+	onToggleVisibility,
+	onMoveUp,
+	onMoveDown,
 }: {
-	field: FieldItemProps;
-	fields: Field< any >[];
-	view: View;
-	onChangeView: ( view: View ) => void;
+	field: NormalizedField< any >;
+	isVisible: boolean;
+	isFirst?: boolean;
+	isLast?: boolean;
+	canMove?: boolean;
+	onToggleVisibility?: () => void;
+	onMoveUp?: () => void;
+	onMoveDown?: () => void;
 } ) {
-	const visibleFieldIds = getVisibleFieldIds( view, fields );
+	const focusVisibilityField = () => {
+		// Focus the visibility button to avoid focus loss.
+		// Our code is safe against the component being unmounted, so we don't need to worry about cleaning the timeout.
+		// eslint-disable-next-line @wordpress/react-no-unsafe-timeout
+		setTimeout( () => {
+			const element = document.querySelector(
+				`.dataviews-field-control__field-${ field.id } .dataviews-field-control__field-visibility-button`
+			);
+			if ( element instanceof HTMLElement ) {
+				element.focus();
+			}
+		}, 50 );
+	};
 
 	return (
-		<Item key={ id }>
+		<Item>
 			<HStack
 				expanded
-				className={ `dataviews-field-control__field dataviews-field-control__field-${ id }` }
+				className={ `dataviews-field-control__field dataviews-field-control__field-${ field.id }` }
+				justify="flex-start"
 			>
-				<span>{ label }</span>
+				<span className="dataviews-field-control__icon">
+					{ ! canMove && ! field.enableHiding && (
+						<Icon icon={ lock } />
+					) }
+				</span>
+				<span className="dataviews-field-control__label">
+					{ field.label }
+				</span>
 				<HStack
 					justify="flex-end"
 					expanded={ false }
 					className="dataviews-field-control__actions"
 				>
-					{ view.type === LAYOUT_TABLE && isVisible && (
+					{ isVisible && (
```

Comment:
I think moving fields could work in grid & list layouts, so leaving those controls visible would be good. Though, right now, they don't work, so this is what happens:

https://github.com/user-attachments/assets/8f4c7678-c75d-49f3-9a10-6f891473022f

If we implement moving the fields in list&grid layouts, it helps us making the view config layout-agnostic. That's a win.

Reply:
I don't think we should allow moving the special fields in either layout, specifically to stay consistent. In "table" it doesn't make sense to put "media" after "title" for instance (same for list I think). So instead of trying to guess what can work where. It's better to be opinionated here. This is the main goal of this PR.

Reply:
> I don't think we should allow moving the special fields in either layout, specifically to stay consistent. In "table" it doesn't make sense to put "media" after "title" for instance (same for list I think). So instead of trying to guess what can work where. It's better to be opinionated here. This is the main goal of this PR.

I agree and that's not what I suggested :) My point was about the other fields: in list & grid the moving controls are visible but they do nothing to the layout. Either we make them invisible or update the layout to actually move the fields (and I prefer the later).

Reply:
>  moving controls are visible but they do nothing to the layout.

Oh I didn't know that. I just assumed that they worked because it was "obvious" to me that they should work :P 
Happy to address that in a follow-up.